specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+        "operationId": "Policies_Update",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PolicyNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "description": "CdnWebAplicationFirewallPolicy parameters to be patched.",
+            "in": "body",
+            "name": "cdnWebApplicationFirewallPolicyPatchParameters",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicyPatchParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          },
+          "202": {
+            "description": "Accepted. The request has been accepted for processing and the operation will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/CdnWebApplicationFirewallPolicy"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./cdn.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Creates specific policy": {
+            "$ref": "./examples/WafPatchPolicy.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      },
+      "delete": {
+        "tags": [
+          "WebApplicationFirewallPolicies"
+        ],
+        "description": "Deletes Policy",
+        "operationId": "Policies_Delete",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PolicyNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Delete successful."
+          },
+          "204": {
+            "description": "No Content. The request has been accepted but the policy was not found."
+          }
+        },
+        "x-ms-examples": {
+          "Delete protection policy": {
+            "$ref": "./examples/WafPolicyDelete.json"
+          }
+        },
+        "x-ms-long-running-operation": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Cdn/CdnWebApplicationFirewallManagedRuleSets": {
```

Comment:
Can you explain what this resource is? It is not clear to me it's relationship with the other new resources.

Reply:
I have already had the ARM review in person with the Arm API board where I explained all the context etc. about this resource.
It would be very inefficient to do it all over again especially via comments.

Reply:
@sushantsingh in cases where you would not like to explain again, if you can point to a design doc or something that explains a bit more, that would be super helpful.

Reply:
Sure I can do that for the future pull requests, for now I do not have this document and I am working hard to get the service ready for the release date.

Reply:
The jist of the change is included in the pull request description.